internal/libs/typeconv/address.go

```diff
@@ -4,27 +4,56 @@ import (
 	"encoding/hex"
 	"fmt"
 	"strings"
+
+	"github.com/gagliardetto/solana-go"
+	chainsel "github.com/smartcontractkit/chain-selectors"
 )
 
 // AddressBytesToString converts the given address bytes to a string
 // based upon the given chain selector's chain family.
-// TODO: only EVM supported for now, fix this.
 func AddressBytesToString(addr []byte, chainSelector uint64) string {
-	// TODO: not EIP-55. Fix this?
-	return "0x" + hex.EncodeToString(addr)
+	family, err := chainsel.GetSelectorFamily(chainSelector)
+	if err != nil {
+		panic(err)
+	}
+
+	switch family {
+	case chainsel.FamilyEVM:
+		// TODO: not EIP-55. Fix this?
+		return "0x" + hex.EncodeToString(addr)
```

Comment:
Just use hexutil.Encode(addr)
It encodes the bytes as a hex string with 0x prefix

Its a geth library from:
"github.com/ethereum/go-ethereum/common/hexutil"

Reply:
@prashantkumar1982 this probably needs to be another service.